### PR TITLE
feat: Create user interests when adding collected artists

### PR DESCRIPTION
**Pull request by olerichter00**
### Description

This PR adds a mutation in order to create user interests when adding collected artists.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Create user interests when adding collected artists - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e0e950cb8f9bc521ceae29787b7f742e808451ad
</p>

File: src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtists.tsx [hunk 1]
```diff
@@ -2,24 +2,59 @@ import { Button, Flex, Spacer, Text, useScreenDimensions } from "@artsy/palette-
...
 export const MyCollectionAddCollectedArtists: React.FC<{}> = () => {
+  const { bottom } = useScreenDimensions().safeAreaInsets
+  const [createUserInterests] = useCreateUserInterests()
+
   const artistIds = MyCollectionAddCollectedArtistsStore.useStoreState((state) => state.artistIds)
 
   const handleSubmit = () => {
-    // Save collected artists
-    // Save personal artists
+    // TODO: Save personal artists
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtists.tsx:18]: **suggestion:** should we also maybe add a `loading` flag to the state and show a spinner while we're saving the user interests for a better UX.

File: src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtists.tests.tsx [hunk 2]
```diff
@@ -1,5 +1,77 @@
...
+import { setupTestWrapper } from "app/utils/tests/setupTestWrapper"
+
 describe("MyCollectionAddCollectedArtists", () => {
-  it("does something", () => {
-    // Do something
+  const { renderWithRelay } = setupTestWrapper({
+    Component: MyCollectionAddCollectedArtistsScreen,
+  })
+
+  it("renders MyCollectionAddCollectedArtists", async () => {
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtists.tests.tsx:13]: 👏

**Review by MounirDhahri — COMMENTED** [review]
LGTM, I left a small comment that was not captured by the ticket. i hope you agree with it. It would be great if you do it to also add a screen recording

File: src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtists.tsx (hunk 1 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtists.tsx:18]: Good idea 👍  If ok, I'll do it in a follow-up PR when working on `// TODO: Save personal artists`.

Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtists.tsx:18]: oki 🙏

**Review by MounirDhahri — APPROVED** [review]

**Pull request merged**